### DISCUSSION TITLE
Early access group feedback

### DISCUSSION THREAD

**Discussion by marekrogala** [Appsilon/shiny.fluent#24] [General]
Hey there!

A couple weeks ago we shared with you the early access version of shiny.fluent. We're especially interested in how you found working with the package overall, what worked well for you, and what you think could be improved. Please post your feedback below.

**What are top 3 things that worked well for you and top 3 things that should be improved in your opinion?**

**Comment by kyle-power** [comment]
Hey @marekrogala !
After playing with the `{shiny.fluent}` API over the past four weeks, I have found that the top three things that worked well for me were...

1. Replicable examples from https://appsilon.github.io/shiny.fluent/articles/st-sales-reps-dashboard.html. This was the first and most critical step to begin using this package, especially given that `{shiny.fluent}` is a very low-level API. Tons of optional parameters and custom styling available that make it unique from other `{shiny}` packages that offer a more _cookie-cutter_ dashboard implementation
2. The combination of `{shiny.fluent}` [documentation](https://appsilon.github.io/shiny.fluent/reference/index.html) in addition to Microsoft `Fluent UI` [documentation](https://developer.microsoft.com/en-us/fluentui#/) make it really easy to start using the basic objects and then take it one step further with optional parameters
3. `{shiny.fluent}` integrates very well with other packages, such as `{golem}` and `{shiny.router}` , which support additional functionality and production deployment

A few things like I would like to see improved are...

1. More built-in styling and default parameters. There are several functions manually defined in the reference articles, such as `makeCard()` and `makePage()`. Since these are more or less required functions, I would imagine that they be built into the base package as available functions that can be further customized if desired. Other "helper"" functions would also be more beneficial if built into the base package - for example, `DetailsList()` requires column names and data to be specified separately, along with a little formatting, ie. a list of lists. I have been using the following function to accomplish this, but would like to see it added as a helper function
```r
makeCols <- function(data){
  tibble(fieldName = names(data), 
         name = names(data), 
         key = names(data), 
         isResizable = TRUE, 
         minWidth = 125) %>%
    group_split(row_number(), .keep = FALSE) %>%
      purrr::map(as.list)
}
```
2. More detailed [function references](https://appsilon.github.io/shiny.fluent/reference/index.html). Most objects have 20 or 30 optional parameters, but since the base `Fluent UI` is mostly `javascript`, it can be difficult to interpret what variable type to provide in the parameter, ie. integer, string, list, list of lists, etc. Perhaps just more extensive examples in the component reference
3. Dynamic sizing/scaling. I frequently struggle with object sizes - column width, button width, etc. Viewport width/height don't seem to work very well so the extremely large and small windows distort the page quite significantly. The same goes for `DetailsList()` which does not automatically adjust the column width to display the content

Overall, I am really enjoying this package and the professional theme that it provides. I am very excited to see `{shiny.fluent}` continue to mature

**Comment by cflagg** [comment]
Hi, I am just discovering Appsilon's shiny packages and am very excited to start trying them out for some internal dashboards I maintain. 

My main question is around where to start: what are the primary differences between `shiny.fomantic` and `shiny.fluent`, or why should one start with one framework versus the other? 

Thanks.

**Comment by PhilSmith26** [comment]
I want to try shiny.fluent but when I tried to install it I received the message "package ‘shiny.fluent’ is not available for this version of R". I am running version 4.1.0 (2021-05-18). What version of R must I use to run shiny.fluent?

**Reply by PhilSmith26** [reply]
I found out that to install shiny.fluent you need to do the following:
remotes::install_github("Appsilon/shiny.react")
remotes::install_github("Appsilon/shiny.fluent")
